### ISSUE TITLE
Change default layout to include exceptions

### ISSUE THREAD

**Issue by StevenLiekens** [NLog/NLog#1292]
The default layout should include a renderer for exceptions.

``` powershell
${longdate}|${level:uppercase=true}|${logger}|${message}|${exception:format=ToString}
```

Relates to #1290 and #1291

**Comment by 304NotModified** [comment]
Thanks for reporting this!

I agree the writing the exception on default is a good suggestion. But I'm doubting between  `${exception:format=ToString}` and `${exception}` - so writing the inner exception or not. The innerException can be pretty long.

**Comment by StevenLiekens** [comment]
I believe that `${exception}` is the same as `${exception:format=message}`. I don't think that's very useful. There should at least be a stack trace of the outermost exception.

**Comment by StevenLiekens** [comment]
After reading #1034, it seems that this is a better default
```powershell
${longdate}|${level:uppercase=true}|${logger}|${message:withException=true}
```

**Comment by 304NotModified** [comment]
note:

- this is also in the xml docs (lot of places)
- we need maybe a exception seperator, too avoid newlines

suggested:  (by @snakefoot - see https://github.com/NLog/NLog/issues/2503#issuecomment-434073043)

     ${longdate}|${level:uppercase=true}|${logger}|${message:withException=true:exceptionSeparator=|}

**Comment by 304NotModified** [comment]
@snakefoot I see it was confused in #2503 with ExceptionLayoutRenderer.Separator

**Comment by snakefoot** [comment]
> I think #1292 (comment) is basically the same and more clear IMO

But it always inserts `|` after the `${message}`, which is not the case with my suggestion

**Comment by 304NotModified** [comment]
Ow yes, you're right.

Maybe we could add another renderer so it's clearer to communicate? Or do you think https://github.com/NLog/NLog/issues/1292#issuecomment-489212731 is easy to understand? (i'm afraid we have to explain it every time)

**Comment by snakefoot** [comment]
My personal opinion is that this is much easier to the eye:

`${longdate}|${level:uppercase=true}|${logger}|${message}|${exception:format=ToString}`

And is actually the Layout that I use normally use. But if NLog 5.0 is meant as minimal breaking changes, then it probably not that good.

**Comment by 304NotModified** [comment]
FYI  We need to fix in NLog 5 that the exception isn't logged by default. That's an issue for ages. It's breaking and so this is the right moment

**Comment by 304NotModified** [comment]
If there aren't exceptions, then this is more backwards compatible: https://github.com/NLog/NLog/issues/1292#issuecomment-489212731

**Comment by snakefoot** [comment]
One could also consider changing `exceptionSeparator` to be `|` by default (So it matches the default-layout and it would be a very little breaking change as it will only have effect when `withException=true`). Then it will be like this:

`${longdate}|${level:uppercase=true}|${logger}|${message:withException=true}`

Other suggestion could be changing `${exception}` to have `format=ToString` by default. Since most people expects `${exception}` to include everything, and not just the exception-message. But it would be a bigger breaking change.

**Comment by 304NotModified** [comment]
> One could also consider changing exceptionSeparator to be | by default

Sounds good to me!

> Other suggestion could be changing ${exception} to have format=ToString by default

Also interesting. Imo it's indeed a better default. I think it's important to have good defaults.

**Comment by snakefoot** [comment]
@304NotModified  Has this been resolved with #3458 ? (So part of NLog 5.0)

**Comment by 304NotModified** [comment]
Not yet. This is about the default for Layout
